hg-fast-export.py

```diff
@@ -128,6 +131,49 @@ def export_file_contents(ctx,manifest,files,hgtags,encoding='',plugins={}):
   count=0
   max=len(files)
   for file in files:
+    if submodule_mappings and ctx.substate and file == ".hgsubstate":
+      # Remove all submodules as we don't detect deleted submodules properly
+      # in any other way. We will add the ones not deleted back again below.
+      for module in submodule_mappings.keys():
+        wr('D %s' % module)
+
+      # Read .hgsubstate file in order to find the revision of each subrepo
+      data = ctx.filectx(file).data()
+      subHashes = {}
+      for line in data.split('\n'):
+        if line.strip() == "":
+          continue
+        cols= line.split(' ')
+        subHashes[cols[1]]=cols[0]
+
+      gitmodules=""
+      # Create the .gitmodules file and all submodules
+      for name in ctx.substate:
+        if submodule_mappings[name]=='SKIP':
```

Comment:
What about a subrepo named `SKIP`? Why not require quoting like the new mapping format, that way filenames can't clash with the keyword.

Reply:
Ok, if I understand correct the format should then be:
`"<mercurial subrepo folder>"="<git submodule name>"|<SKIP>`
E.g.
```
"subrepo/subrepo1"="../submodule1"
"subrepo/subrepo2"="../submodule2"
"subrepo/subrepo3"=SKIP
```
It seems like the load_mapping() function does not support this (it requires quotes). What about:
`"subrepo/subrepo3"="^SKIP"`

Reply:
Or perhaps just an empty string:
`"subrepo/subrepo3"=""`

Reply:
`"^SKIP"` is not a solution as it too could be a (although unlikely) filename. En empty string would probably work but feels hacky, why not extend `load_mapping()` to map to None when the keyword `SKIP` is used?

Reply:
I modified the regexp in `load_mapping()` and it works, but the resulting regexp is a bit hard to read. I will drop the "SKIP" functionality for now as it only covers a corner case and the increased code complexity is probably not worth it.